server/src/session.ts

```diff
@@ -378,19 +374,54 @@ export class Session {
   listen() {
     this.connection.listen();
   }
+
+  /**
+   * Determine if the specified `project` is Angular, and disable the language
+   * service if not.
+   * @param project
+   */
+  private checkIsAngularProject(project: ts.server.Project) {
+    const NG_CORE = '@angular/core/core.d.ts';
+    const {projectName} = project;
+    if (!project.languageServiceEnabled) {
+      const msg = `Language service is already disabled for ${projectName}. ` +
+          `This could be due to non-TS files that exceeded limit (${
```

Comment:
```suggestion
          `This could be due to non-TS files that exceeded the size limit (${ts.server.maxProgramSizeForNonTsFiles} bytes).`
```

I think we should be as clear as possible; a large number with no annotation might be ambiguous. LGTM otherwise.